nautobot/nautobot_data_validation_engine/custom_validators.py

```diff
@@ -193,12 +198,19 @@ def get_classes_from_git_repo(repo: GitRepository):
     """Get list of DataComplianceRule classes found within the custom_validators folder of the given repo."""
     ensure_git_repository(repo, head=repo.current_head)
     class_list = []
-    for importer, discovered_module_name, _ in pkgutil.iter_modules([f"{repo.filesystem_path}/custom_validators"]):
-        if discovered_module_name in sys.modules:
-            del sys.modules[discovered_module_name]
-        module = importer.find_module(discovered_module_name).load_module(discovered_module_name)
-        for _, compliance_class in inspect.getmembers(module, is_data_compliance_rule):
-            class_list.append(compliance_class)
+    custom_validators_path = pathlib.Path(f"{repo.filesystem_path}/custom_validators")
+
+    for module_path in custom_validators_path.glob("*.py"):  # Only load Python files
+        module_name = module_path.stem  # Extract the filename without ".py" extension
```

Comment:
Let's compare this to how Job loading is done from a Git repo. In particular I'm concerned that this approach doesn't *look* like it namespaces the loaded modules under the Git repo slug, and as such may potentially clobber system modules (`nautobot.py -> sys.modules["nautobot"]`)? It also may not support relative imports amongst files in the same directory which is something we know users will want to do.

Reply:
The job loading code is quite a beast to understand, in the doc string it literally says "If you find yourself writing new code that uses this method, please pause and reconsider your life choices." 

I think this [commit](https://github.com/nautobot/nautobot/pull/7074/commits/59bebd441521423b958e9d896e1fba74fe930efd) addresses the potential clobbering of system modules concern by adding namespacing the loaded modules with the Git repo slug.

I am a little stuck on adding support for relative imports.